run `minikube start` with --image-mirror-country=cn failed

When i use the **minikube 1.8.2** to start , it failed, But the 1.8.1 can work.

**The exact command to reproduce the issue**:

```
minikube start --cpus=4 --memory=4g --download-only=true --driver=kvm2 --image-mirror-country=cn --alsologtostderr -v=7
```
**The full output of the command that failed**:<details>
```

panic: semver: Parse(v1.17.3): Invalid character(s) found in major number "v1"

goroutine 1 [running]:
github.com/blang/semver.MustParse(0xc0007ee036, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/pkg/mod/github.com/blang/semver@v3.5.0+incompatible/semver.go:319 +0x1c1
k8s.io/minikube/cmd/minikube/cmd.generateCfgFromFlags(0x2a45e60, 0xc0007ee036, 0x7, 0x1a6b53a, 0x4, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/app/cmd/minikube/cmd/start.go:825 +0x335
k8s.io/minikube/cmd/minikube/cmd.runStart(0x2a45e60, 0xc000148a80, 0x0, 0x8)
	/app/cmd/minikube/cmd/start.go:315 +0x3be
github.com/spf13/cobra.(*Command).execute(0x2a45e60, 0xc000148900, 0x8, 0x8, 0x2a45e60, 0xc000148900)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0x2a44f60, 0x0, 0x1, 0xc000637bc0)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
k8s.io/minikube/cmd/minikube/cmd.Execute()
	/app/cmd/minikube/cmd/root.go:108 +0x6a4
main.main()
	/app/cmd/minikube/main.go:66 +0xea

```

</details>

**The output of the `minikube logs` command**:<details>



</details>

**The operating system version**:

Ubuntu 18.04.4 LTS

--- a/cmd/minikube/cmd/start.go
+++ b/cmd/minikube/cmd/start.go
@@ -822,7 +822,7 @@ func generateCfgFromFlags(cmd *cobra.Command, k8sVersion string, drvName string)
 	repository := viper.GetString(imageRepository)
 	mirrorCountry := strings.ToLower(viper.GetString(imageMirrorCountry))
 	if strings.ToLower(repository) == "auto" || mirrorCountry != "" {
-		found, autoSelectedRepository, err := selectImageRepository(mirrorCountry, semver.MustParse(k8sVersion))
+		found, autoSelectedRepository, err := selectImageRepository(mirrorCountry, semver.MustParse(strings.TrimPrefix(k8sVersion, version.VersionPrefix)))
 		if err != nil {
 			exit.WithError("Failed to check main repository and mirrors for images for images", err)
 		}

--- a/cmd/minikube/cmd/start_test.go
+++ b/cmd/minikube/cmd/start_test.go
@@ -70,6 +70,50 @@ func TestGetKuberneterVersion(t *testing.T) {
 	}
 }
 
+func TestMirrorCountry(t *testing.T) {
+	// Set default disk size value in lieu of flag init
+	viper.SetDefault(humanReadableDiskSize, defaultDiskSize)
+
+	k8sVersion := constants.DefaultKubernetesVersion
+	var tests = []struct {
+		description     string
+		k8sVersion      string
+		imageRepository string
+		mirrorCountry   string
+		cfg             *cfg.ClusterConfig
+	}{
+		{
+			description:     "image-repository none, image-mirror-country none",
+			imageRepository: "",
+			mirrorCountry:   "",
+		},
+		{
+			description:     "image-repository auto, image-mirror-country none",
+			imageRepository: "auto",
+			mirrorCountry:   "",
+		},
+		{
+			description:     "image-repository auto, image-mirror-country china",
+			imageRepository: "auto",
+			mirrorCountry:   "cn",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			cmd := &cobra.Command{}
+			viper.SetDefault(imageRepository, test.imageRepository)
+			viper.SetDefault(imageMirrorCountry, test.mirrorCountry)
+			config, _, err := generateCfgFromFlags(cmd, k8sVersion, "none")
+			if err != nil {
+				t.Fatalf("Got unexpected error %v during config generation", err)
+			}
+			// the result can still be "", but anyway
+			_ = config.KubernetesConfig.ImageRepository
+		})
+	}
+}
+
 func TestGenerateCfgFromFlagsHTTPProxyHandling(t *testing.T) {
 	// Set default disk size value in lieu of flag init
 	viper.SetDefault(humanReadableDiskSize, defaultDiskSize)

